opentelemetry-otlp/src/retry.rs

```diff
@@ -0,0 +1,147 @@
+use std::future::Future;
+use std::time::{Duration, SystemTime};
+use opentelemetry::otel_warn;
+use tokio::time::sleep;
```

Comment:
`tokio` runtime won't be available here. Should we move this code under `exporter/tonic`, or if we want to keep it generic, one option could be to make the delay function configurable as a function argument.

Reply:
The idea was the retry logic should be generic and used across all the OTLP exporters. Going to try that approach first and try to avoid the direct tokio dependency.